common/functions/src/scalars/maths/trigonometric.rs

```diff
@@ -63,8 +64,22 @@ impl Function for TrigonometricFunction {
         1
     }
 
-    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
-        Ok(DataType::Float64)
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if is_numeric(&args[0]) || args[0] == DataType::String {
+            Ok(match &args[0] {
+                DataType::Int8 => DataType::UInt8,
```

Comment:
maybe the return value of trigonometric functions are always float64? sin(1) -> 0.8414709848078965

Reply:
Yes，a mistake..